filebeat/prospector/prospector_log.go

```diff
@@ -94,24 +94,35 @@ func (l *Log) Run() {
 	if l.config.CleanRemoved {
 		for _, state := range l.Prospector.states.GetStates() {
 			// os.Stat will return an error in case the file does not exist
-			_, err := os.Stat(state.Source)
+			stat, err := os.Stat(state.Source)
 			if err != nil {
```

Comment:
Yea that's probably the best we can do. `os.PathError` is all that can return right now but you never now..